api/src/main/java/org/apache/iceberg/UpdateStatistics.java

```diff
@@ -27,9 +27,22 @@ public interface UpdateStatistics extends PendingUpdate<List<StatisticsFile>> {
    * the snapshot if any exists.
    *
    * @return this for method chaining
+   * @deprecated since 1.8.0, will be removed 1.9.0 or 2.0.0, use {@link
+   *     #setStatistics(StatisticsFile)}.
    */
+  @Deprecated
   UpdateStatistics setStatistics(long snapshotId, StatisticsFile statisticsFile);
 
+  /**
+   * Set the table's statistics file for given snapshot, replacing the previous statistics file for
+   * the snapshot if any exists.
+   *
+   * @return this for method chaining
+   */
```

Comment:
nit: should we mention in the docstring that the snapshot id is coming from the statistics file?

Reply:
I like this - added a sentence to the comment.